### ISSUE TITLE
Lint `f` strings better!

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#1239]
# Rule request

## Thesis

Currently we just do not allow people to use `f` strings.
But, what if our user just ignores this violation? Now it is possible to do very bad things inside `f` strings and we do not cover it at all.

I suggest to add `TooComplexFormattedStringViolation` that will be throw in case of:
0. `f'{one_name}'` should raise: because we only a single name inside `f` string
1. Too complex expression inside `f` string: like calling a lot of function, making conditions, etc
2. Too many expressions used in `f` string
3. `f` strings that dont have any values inside like: `f'abc'`
4. Things like `assert f'a {value}'` should raise proper violations (`WPS444` in this case), but `assert f'{value}'` should not

This is just a discussion at the moment.

**Comment by JazabC** [comment]
@sobolevn Can I take this issue?

**Comment by sobolevn** [comment]
Sure, but I am not sure it is ready to be implemented yet. 
Because I don't know complexity metrics to implement.

@AlwxSin do you want to share your thoughts on this one?

**Comment by AlwxSin** [comment]
From my experience, one can easily read code like this
```python
f'smth {value}'
f'smth {dict_value["key"]}'
f'smth {list_value[0]}'
f'smth {user.get_full_name()}'
def get_full_name(self):
    return f'{self.first_name} {self.second_name} {self.last_name}'.strip()
```

Anything more complex is hard to read like `f'{reverse('url-name')}?{"&".join("user="+uid for uid in user_ids)}'`

So:
* value of a variable itself
* value of a dict key by key lookup
* value of a list item by index lookup
* return value of a function or a method without arguments

**Comment by ruwaid4** [comment]
Hi again @sobolevn, we would like to take this one on.

**Comment by sobolevn** [comment]
Thanks a lot!

**Comment by ruwaid4** [comment]
@sobolevn @AlwxSin we were thinking about whitelisting the cases mentioned in this thread. Are there any new cases you've thought of that should be allowed?

**Comment by AlwxSin** [comment]
@ruwaid4 no more than I said before 🤷

**Comment by ruwaid4** [comment]
@sobolevn Just to clarify, FormattedStringViolation should NOT be removed but TooComplexFormattedStringViolation should be added in case a user decides to ignore the former violation?
